### ISSUE TITLE
Package breaks with Flutter 1.23

### ISSUE THREAD

**Issue by IoanaAlexandru** [JonasWanke/timetable#55]
**Describe the bug**

The timetable widget fails building after upgrading from `1.22.0-12.4.pre` to `1.23.0-18.1.pre`, on the beta channel. This is the error:

<details><summary>Stack trace</summary>

```
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following _CastError was thrown during performLayout():
Null check operator used on a null value

The relevant error-causing widget was:
  Scrollable
  file:///C:/Users/sako_/flutter/.pub-cache/hosted/pub.dartlang.org/timetable-0.2.8/lib/src/date_page_view.dart:46:12

When the exception was thrown, this was the stack:
#0      ScrollPosition.pixels (package:flutter/src/widgets/scroll_position.dart:166:31)
#1      ScrollController.offset (package:flutter/src/widgets/scroll_controller.dart:116:33)
#2      ScrollController.debugFillDescription (package:flutter/src/widgets/scroll_controller.dart:270:45)
#3      ScrollController.toString (package:flutter/src/widgets/scroll_controller.dart:247:5)
#4      _StringBase._interpolate (dart:core-patch/string_patch.dart:848:19)
#5      _LinkedScrollPosition.debugFillDescription (package:timetable/src/utils/scrolling.dart:316:36)
#6      ViewportOffset.toString (package:flutter/src/rendering/viewport_offset.dart:234:5)
#7      DiagnosticsProperty.valueToString (package:flutter/src/foundation/diagnostics.dart:2740:60)
#8      DiagnosticsProperty.toDescription (package:flutter/src/foundation/diagnostics.dart:2754:21)
#9      TextTreeRenderer._debugRender (package:flutter/src/foundation/diagnostics.dart:1193:32)
#10     TextTreeRenderer.render (package:flutter/src/foundation/diagnostics.dart:1126:14)
#11     TextTreeRenderer._debugRender (package:flutter/src/foundation/diagnostics.dart:1303:39)
#12     TextTreeRenderer.render (package:flutter/src/foundation/diagnostics.dart:1126:14)
#13     TextTreeRenderer._debugRender (package:flutter/src/foundation/diagnostics.dart:1317:39)
#14     TextTreeRenderer.render (package:flutter/src/foundation/diagnostics.dart:1126:14)
#15     FlutterError.dumpErrorToConsole (package:flutter/src/foundation/assertions.dart:961:11)
#16     TestWidgetsFlutterBinding._runTest.<anonymous closure> (package:flutter_test/src/binding.dart:662:24)
#17     FlutterError.reportError (package:flutter/src/foundation/assertions.dart:1093:15)
#18     RenderObject._debugReportException (package:flutter/src/rendering/object.dart:1311:18)
#19     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
#20     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#21     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#22     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#23     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#24     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#25     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#26     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#27     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#28     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#29     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#30     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#31     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#32     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#33     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#34     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#35     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#36     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#37     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#38     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#39     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#40     RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:831:17)
#41     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#42     RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:266:14)
#43     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#44     _RenderSingleChildViewport.performLayout (package:flutter/src/widgets/single_child_scroll_view.dart:544:14)
#45     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#46     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#47     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#48     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#49     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#50     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#51     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#52     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#53     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#54     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#55     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#56     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#57     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#58     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#59     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#60     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#61     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#62     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#63     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#64     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#65     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#66     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#67     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#68     _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:353:14)
#69     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#70     RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:831:17)
#71     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#72     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:565:15)
#73     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#74     MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:171:12)
#75     _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:893:7)
#76     MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:243:7)
#77     RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:402:14)
#78     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#79     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#80     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#81     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#82     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1306:11)
#83     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#84     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#85     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#86     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#87     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#88     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#89     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#90     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:111:14)
#91     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#92     RenderSliverMultiBoxAdaptor.insertAndLayoutChild (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:488:13)
#93     RenderSliverFixedExtentBoxAdaptor.performLayout (package:flutter/src/rendering/sliver_fixed_extent_list.dart:242:17)
#94     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#95     RenderSliverEdgeInsetsPadding.performLayout (package:flutter/src/rendering/sliver_padding.dart:139:12)
#96     _RenderSliverFractionalPadding.performLayout (package:flutter/src/widgets/sliver_fill.dart:168:11)
#97     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
#98     RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:508:13)
#99     RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1566:12)
#100    RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1475:20)
#101    RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1634:7)
#102    PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:884:18)
#103    AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:1107:23)
#104    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:309:5)
#105    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
#106    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
#107    AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:974:9)
#110    TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#111    AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:961:27)
#112    WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:640:23)
#123    FakeAsync.flushMicrotasks (package:fake_async/fake_async.dart:193:32)
#124    AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1208:17)
#125    AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1195:35)
(elided 27 frames from dart:async and package:stack_trace)
```

</details>

**Environment:**

- Device:  Pixel 4
- OS version:  Android 10.0
- Package version:  0.2.8

**Comment by 7Love7code7** [comment]
Hello friends.
timetable package is really awesome.
But unfortunately, I got this same error after update flutter version.
Is there any solution for this?
Thanks in advance.

**Comment by IoanaAlexandru** [comment]
A temporary solution is to just run `flutter downgrade` and avoid upgrading until the package is fixed. @JonasWanke could you give us an estimate as to when this could be done?

**Comment by JonasWanke** [comment]
This is now fixed in [v0.2.9](https://github.com/JonasWanke/timetable/releases/tag/v0.2.9).

**Comment by 7Love7code7** [comment]
Super!Great thanks Sent from Mail for Windows 10 From: Jonas WankeSent: Monday, October 26, 2020 11:44 AMTo: JonasWanke/timetableCc: 7Love7code7; CommentSubject: Re: [JonasWanke/timetable] Package breaks with Flutter 1.23 (#55) This is now fixed in v0.2.9.—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
